File: benchmark/c/cpw/squared_triangular_number_sum_cubes/squared_triangular_number_sum_cubes.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int squared_triangular_number_sum_cubes ( int s ) ;
int squared_triangular_number_sum_cubes ( int s ) {
  int sum = 0;
  for ( int n = 1;
  sum < s;
  n ++ ) {
    sum += n * n * n;
    if ( sum == s ) return n;
  }
  return - 1;
}

File: translation_result/else_bms/fail/c/cpw/squared_triangular_number_sum_cubes/gemini/Hinted/squared_triangular_number_sum_cubes.rs

fn squared_triangular_number_sum_cubes(s: i32) -> i32 {
    let mut sum = 0;
    let mut n = 1;
    while sum < s {
        sum = sum.wrapping_add(n * n * n);
        if sum == s {
            return n;
        }
        n += 1;
    }
    -1
}
